### PR TITLE
[Service Connector] `az webapp/containerapp/spring connection create/update`: Store configurations in App Config

### PR DESCRIPTION
**Pull request by wchigit**
**Related command**
az webapp/containerapp/spring connection create/update <targer-service-type> --appconfig-id <app-config-id>

**Description**
This PR supports writing configurations that Service Connector provides into App Config. A connection from source resource to App Config is automatically built if it doe not already exist to support read/write operations of App Config. 

**Testing Guide**
Create connection: az webapp connection create storage-blob --connection {connection-name} --source-id {source-id} --target-id {target-id} --secret --client-type python --appconfig-id {appconfig-id}
After creation, check the key-value pair stored in Appp Config: az appconfig kv list --name {config_store_name} --key {configuration_name}

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Service Connector

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py [hunk 1]
```diff
@@ -434,6 +441,11 @@ def connection_create_func(cmd, client,  # pylint: disable=too-many-locals,too-m
             and auth_info['secret_info']['secret_type'] == 'keyVaultSecretReference':
         raise ValidationError('--vault-id must be provided to use secret-name')
 
+    if app_config_id:
+        client = set_user_token_header(client, cmd.cli_ctx)
```
Comment (ChenTanyi) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py:445]: no need user token now

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py [hunk 2]
```diff
@@ -192,6 +192,11 @@ def add_secret_store_argument(context):
         '--vault-id'], help='The id of key vault to store secret value')
 
 
+def add_configuration_store_argument(context):
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py:195]: only support resource id? what about also support app config name in same resource group?

Comment (wchigit) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py:195]: Using source resource group as default is not supported. We may support this in the future.

**Review by ChenTanyi — APPROVED** [review]

**Comment by wchigit** [comment]
@kairu-ms Could you please help review this PR?

**Comment by wchigit** [comment]
@kairu-ms Kind reminder.

**Review by zhoxing-ms — DISMISSED** [review]

**Pull request merged**